code/modules/clothing/chameleon/chameleon_gun.dm

```diff
@@ -54,36 +56,43 @@
 	inhand_x_dimension = gun_to_set.inhand_x_dimension
 	inhand_y_dimension = gun_to_set.inhand_y_dimension
 
+	// We dupe this casing and then delete it at the end, to grab the projectile.
+	var/obj/item/ammo_casing/casing_to_dupe
+
 	if(istype(gun_to_set, /obj/item/gun/ballistic))
 		var/obj/item/gun/ballistic/ball_gun = gun_to_set
+		// We also need to copy the starting magazine for ballistics.
 		var/obj/item/ammo_box/ball_ammo = new ball_gun.spawn_magazine_type(gun_to_set)
```

Comment:
Why do we even need to instantiate this? Can't we just use `initial`?
```dm
var/obj/item/casing_type = initial(ball_gun.spawn_magazine_type.ammo_type)
casing_to_dupe = new casing_type(src)
```

Reply:
spawn_magazine_type is a proc???

Reply:
its both a proc and a variable holy shit